#goal
float32[] target_xs
float32[] target_ys
float32 stopping_distance
---
#result
float32 time_elapsed
---
#feedback
float32 current_target_x
float32 current_target_y
string current_state

# it might make more sense to make target be a single coordinate
# pair and move the logic of going through each target to the client
